progress/bar.go

```diff
@@ -50,44 +57,49 @@ func (b *Bar) String() string {
 
 	var pre, mid, suf strings.Builder
 
-	if b.message != "" {
+	if len(b.message) > 0 {
 		message := strings.TrimSpace(b.message)
 		if b.messageWidth > 0 && len(message) > b.messageWidth {
 			message = message[:b.messageWidth]
 		}
 
 		fmt.Fprintf(&pre, "%s", message)
-		if b.messageWidth-pre.Len() >= 0 {
-			pre.WriteString(strings.Repeat(" ", b.messageWidth-pre.Len()))
+		if padding := b.messageWidth - pre.Len(); padding > 0 {
+			pre.WriteString(strings.Repeat(" ", padding))
 		}
 
 		pre.WriteString(" ")
 	}
 
 	fmt.Fprintf(&pre, "%3.0f%% ", math.Floor(b.percent()))
-	fmt.Fprintf(&suf, "(%s/%s", format.HumanBytes(b.currentValue), format.HumanBytes(b.maxValue))
 
-	stats := b.Stats()
-	rate := int64(stats.rate)
-	if rate > 0 {
-		fmt.Fprintf(&suf, ", %s/s", format.HumanBytes(rate))
-	}
+	suf.WriteRune('(')
 
-	fmt.Fprintf(&suf, ")")
+	rate := b.rate()
+	remaining := time.Duration(int64(float64(b.maxValue-b.currentValue)/rate)) * time.Second
 
-	elapsed := time.Since(b.started)
-	if b.percent() < 100 && rate > 0 {
-		fmt.Fprintf(&suf, " [%s:%s]", elapsed.Round(time.Second), stats.remaining)
-	} else {
-		fmt.Fprintf(&suf, "        ")
```

Comment:
This whitespace was to keep the bar from expanding dramatically at the end which can be jarring

Reply:
This has the same width as the format string but the real formatted string will be wider since the minimum value is ` [0s:0s]`

Reply:
Yes – it was more a quick and dirty way to keep the space, at the minimum, the size of the format string. 

Ideally we render a fixed length progress bar with variable length stats on the side